### PR TITLE
Fix paraphase

### PR DESCRIPTION
**Pull request by bagnacan**
This PR fixes a bug in the paraphase module.

Paraphase's output vcf/vcf-index files are created only if the provided input BAM file has reads in the analyzed region. When the analyzed region has no reads, paraphase creates no vcf/vcf-index files. For this reason, the paraphase module should zip these files only when they are available.

This PR introduces a check for the availability of these files, as well as a new test to verify that these outputs are indeed optional. Test data needed to be extended to cover for this corner case (see [test-dataset PR here](https://github.com/nf-core/test-datasets/pull/1860)).  
Finally, this PR bumps the versions of _paraphase_ and _samtools_, and modifies the module to use topic channels.

## PR checklist

Relates to #9929 and #10099.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If necessary, include test data in your PR.

### CODE REVIEW

File: modules/nf-core/paraphase/tests/main.nf.test [hunk 1]
```diff
@@ -93,7 +93,44 @@ nextflow_process {
...
+                    process.out.findAll { key, val -> key.startsWith("versions") }
+                    ).match()
+                }
+            )
+        }
+    }
+
+    test("homo_sapiens - [ bam, bai ], no_vcf") {
+
+        config "./nextflow.no_vcf.config"
```
Comment (fellen31) [inline modules/nf-core/paraphase/tests/main.nf.test]: Could you change to use just one nextflow.config and https://nf-co.re/docs/guidelines/components/modules#configuration-of-extargs-in-tests?

File: modules/nf-core/paraphase/tests/main.nf.test [hunk 2]
```diff
@@ -93,7 +93,44 @@ nextflow_process {
...
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert process.out.vcf?.isEmpty() },
+                { assert process.out.vcf_index?.isEmpty() },
```
Comment (fellen31) [inline modules/nf-core/paraphase/tests/main.nf.test]: Can `process.out.vcf` ever be `null`? 

```suggestion
                { assert process.out.vcf.isEmpty() },
                { assert process.out.vcf_index.isEmpty() },
```

File: modules/nf-core/paraphase/tests/main.nf.test [hunk 3]
```diff
@@ -93,7 +93,44 @@ nextflow_process {
...
+                { assert snapshot(
+                    file(process.out.json.get(0).get(1)).readLines()[0..28],
+                    process.out.bam.get(0).get(1).isEmpty(),
```
Comment (fellen31) [inline modules/nf-core/paraphase/tests/main.nf.test]: Shouldn't there still be a bam output? Looks like it in the in the snapshot since `process.out.bam.get(0).get(1).isEmpty()` is `false`. Could we at least snapshot the name if `bam(process.out.bam.get(0).get(1)).getReadsMD5()` doesn't work?

File: modules/nf-core/paraphase/tests/main.nf.test (hunk 2 above)
Comment (bagnacan) [reply modules/nf-core/paraphase/tests/main.nf.test]: Can be removed here. I think I left it there when I was checking paraphase's behavior before updating the module's script.

File: modules/nf-core/paraphase/tests/main.nf.test (hunk 3 above)
Comment (bagnacan) [reply modules/nf-core/paraphase/tests/main.nf.test]: Good catch, I'll change that line.

File: modules/nf-core/paraphase/tests/main.nf.test (hunk 1 above)
Comment (bagnacan) [reply modules/nf-core/paraphase/tests/main.nf.test]: Thanks for the pointer. I didn't know what one _should_ use a single nextflow.config when testing.

However, when I add a `when.params` block:
```
    test("homo_sapiens - [ bam, bai ], no_vcf") {

        //config "./nextflow.no_vcf.config"

        when {
            params {
                args = "--gene RIMBP3B"
            }
            process {
                ...
```
the `gene` parameter does not overwrite the one provided by `./nextflow.config`. In fact, a `PRODH` VCF file is always created.  
Since the point of this test is to show that the process _optionally_ creates VCFs, then the presence of the `PRODH` VCF file invalidates the purpose of this test.  
Am I defining the params block incorrectly? Otherwise, I propose to keep the 2nd `nextflow.no_vcf.config`.  
What do you think?

Comment (fellen31) [reply modules/nf-core/paraphase/tests/main.nf.test]: I think you need to update both the no_vcf test, and the previous tests with `params { ... }` blocks.

Comment (bagnacan) [reply modules/nf-core/paraphase/tests/main.nf.test]: It still does not work if I remove the `config` directive from any/all of the tests (and replace it with a reative `when.params` block).  
Just to be sure, I even commented out every other test, and run the `no_vcf` test alone. It still produces a PRODH VCF file. So I think that the `./nextflow_config` is always loaded unless ovewritten.  
Do you have any recommendation?

Comment (fellen31) [reply modules/nf-core/paraphase/tests/main.nf.test]: Could you push your changes so I can take a look?

Comment (bagnacan) [reply modules/nf-core/paraphase/tests/main.nf.test]: Thanks, @fellen31.
I pushed 3 commits, the last of which tries to address your 1st comment.

File: modules/nf-core/paraphase/tests/nextflow.no_vcf.config
Comment (fellen31) [inline modules/nf-core/paraphase/tests/nextflow.no_vcf.config:1]: Remove this.

File: modules/nf-core/paraphase/tests/main.nf.test (hunk 1 above)
Comment (fellen31) [reply modules/nf-core/paraphase/tests/main.nf.test]: Cool, so what's missing is that you need to update `.nextflow.config` from 

```
process {
  withName: 'PARAPHASE' {
    ext.args = '--gene PRODH'
  }
}
```

to

```
process {
  withName: 'PARAPHASE' {
    ext.args = params.module_args
  }
}
```

so that `--gene PRODH` is not hardcoded, and `params.module_args` can be picked up instead and sent to the module (via ext.args). 

Then use `module_args`, not `ext.args` in main.nf.test‎, e.g.: 

```
params {
    module_args = '--gene RIMBP3B'
  }
```

for your no_vcf tests, and replace it with 

```
params {
    module_args = '--gene PRODH'
  }
```

in the other tests.

Comment (bagnacan) [reply modules/nf-core/paraphase/tests/main.nf.test]: Ah, very nice, I didn't know I could do:

```
process {
  withName: 'PARAPHASE' {
    ext.args = params.module_args
  }
}
```

Thanks for the tip!

File: modules/nf-core/paraphase/tests/main.nf.test
Comment (fellen31) [inline modules/nf-core/paraphase/tests/main.nf.test:7]:
```suggestion
process "PARAPHASE"
config "./nextflow.config"

```

File: modules/nf-core/paraphase/tests/main.nf.test [hunk 4]
```diff
@@ -28,6 +28,9 @@ nextflow_process {
         config "./nextflow.config"
```
Comment (fellen31) [inline modules/nf-core/paraphase/tests/main.nf.test]:
```suggestion
```

File: modules/nf-core/paraphase/tests/main.nf.test [hunk 5]
```diff
@@ -67,6 +70,9 @@ nextflow_process {
...
```
Comment (fellen31) [inline modules/nf-core/paraphase/tests/main.nf.test]:
```suggestion
```

File: modules/nf-core/paraphase/tests/main.nf.test [hunk 6]
```diff
@@ -93,7 +99,47 @@ nextflow_process {
...
+                    ).match()
+                }
+            )
+        }
+    }
+
+    test("homo_sapiens - [ bam, bai ], no_vcf") {
+
+        config "./nextflow.config"
+
```
Comment (fellen31) [inline modules/nf-core/paraphase/tests/main.nf.test]:
```suggestion
```

File: modules/nf-core/paraphase/tests/main.nf.test [hunk 7]
```diff
@@ -174,6 +223,9 @@ nextflow_process {
         config "./nextflow.config"
```
Comment (fellen31) [inline modules/nf-core/paraphase/tests/main.nf.test]:
```suggestion
```

File: modules/nf-core/paraphase/tests/main.nf.test [hunk 8]
```diff
@@ -144,6 +190,9 @@ nextflow_process {
...
```
Comment (fellen31) [inline modules/nf-core/paraphase/tests/main.nf.test]:
```suggestion
```
